File: data/samples/sample_code_fixed.js
```javascript
// Removed hardcoded secret, consider using environment variables instead
// To fix: use a secure method to store and retrieve sensitive information
const password = process.env.PASSWORD;

let x = 10;
let y = 0;

/**
 * Calculate the result of two numbers based on the given operation.
 * 
 * @param {number} x The first number.
 * @param {number} y The second number.
 * @param {string} operation The operation to perform. Can be "add", "subtract", "multiply", or "divide".
 * @returns {number} The result of the operation.
 */
function calculate(x, y, operation) {
    // Improved the operation checks using a switch statement for better readability
    switch (operation) {
        case "add":
            return x + y;
        case "subtract":
            return x - y;
        case "multiply":
            return x * y;
        case "divide":
            // Added a zero check to prevent division by zero
            if (y === 0) {
                throw new Error("Cannot divide by zero");
            }
            return x / y;
        default:
            // Added a default case to handle unknown operations
            throw new Error(`Unknown operation: ${operation}`);
    }
}

/**
 * Process an array of numbers by doubling each non-null value.
 * 
 * @param {number[]} data The array of numbers to process.
 * @returns {number[]} The processed array with doubled values.
 */
function processData(data) {
    const result = [];
    for (let i = 0; i < data.length; i++) {
        // Changed the null check to use the strict !== operator
        if (data[i] !== null) {
            result.push(data[i] * 2);
        }
    }
    return result;
}

// Removed the unused variable to prevent code clutter

console.log(calculate(x, 1, "divide"));  // Avoid dividing by zero
```

File: data/samples/sample_code_fixed.py
import os
import sys

# Removed duplicate import

def calculate(x, y, operation):
    # Improved error handling
    if operation == "add":
        return x + y
    elif operation == "subtract":
        return x - y
    elif operation == "multiply":
        return x * y
    elif operation == "divide":
        # Raise an exception for division by zero
        if y == 0:
            raise ZeroDivisionError("Cannot divide by zero")
        return x / y
    elif operation == "power":
        return x ** y
    else:
        # Raised an exception for invalid operation
        raise ValueError("Invalid operation")

def process_data(data):
    result = []
    # Using enumerate for clearer code
    for i, item in enumerate(data):
        # Using 'is not None' for correct comparison
        if item is not None:
            result.append(item * 2)
    return result

# Removed hardcoded secret, consider using environment variables or secure storage
import dotenv
dotenv.load_dotenv()
password = os.getenv("PASSWORD")

# Removed unused function

# Consider adding main function or guard clause for script execution
if __name__ == "__main__":
    # Example usage
    print(calculate(5, 2, "add"))
    print(process_data([1, 2, None, 4]))